addons/olap/models/report_view.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+from odoo import models, fields, api, _
+
+
+class ReportViewer(models.TransientModel):
+    _name = "olap.report.viewer"
+    _description = """Transient model, custom report can specify a few fields
+    and then use this form to show a pre-filtered list of report records, and 
+    optionally add some extra stuff, like a graph summarising the report 
+    records, or totals.
+    """
+
+    def _default_from_date(self):
+        max_age = datetime.now() - timedelta(days=30)
+        return fields.Datetime.to_string(max_age)
+
+    from_date = fields.Datetime('From:', default=_default_from_date)
```

Comment:
should that be '_default_from_date'?

Reply:
No, the default can be a variety of things, one of which is a function that will be run per-record if you want a default that is not static as we do here:
https://www.odoo.com/documentation/11.0/howtos/backend.html#default-values